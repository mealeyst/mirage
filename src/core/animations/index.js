export { default as FadeInOut } from './FadeInOut'
