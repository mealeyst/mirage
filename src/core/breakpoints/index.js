import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Desktop = styled(({element, children, ...props}) => {
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose.aboveTabletMax`
  display: ${props => props.display};
`}
`

Desktop.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Desktop.defaultProps = {
  display: 'block',
  element: 'div'
}

const Laptop = styled(({element, children, ...props}) => {
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose.aboveLaptop`
  display: ${props => props.display};
`}
`

Laptop.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Laptop.defaultProps = {
  display: 'block',
  element: 'div'
}

const Tablet = styled(({element, children, ...props}) => {
  delete props.laptopDisplay
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose.aboveTablet`
  display: ${props => props.display};
`}
${props => props.theme.breakpointsVerbose.aboveLaptop`
  display: ${props => props.laptopDisplay};
`}
`

Tablet.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Tablet.defaultProps = {
  display: 'block',
  element: 'div',
  laptopDisplay: 'none'
}

const TabletMax = styled(({element, children, ...props}) => {
  delete props.laptopDisplay
  return React.createElement(element, props, children)
})`
  display: none;
  ${props => props.theme.breakpointsVerbose.aboveTabletMax`
    display: ${props => props.display};
  `}
  ${props => props.theme.breakpointsVerbose.aboveLaptop`
    display: ${props => props.laptopDisplay};
  `}
`

TabletMax.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
TabletMax.defaultProps = {
  display: 'block',
  element: 'div',
  laptopDisplay: 'none'
}

const Mobile = styled(({element, children, ...props}) => {
  return React.createElement(element, props, children)
})`
display: ${props => props.display};
${props => props.theme.breakpointsVerbose.aboveTablet`
  display: none;
`}
`

Mobile.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Mobile.defaultProps = {
  display: 'block',
  element: 'div'
}

const Default = styled(({element, children, ...props}) => {
  delete props.displayTarget
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose[props.displayTarget]`
    display: ${props => props.display};
`}
`

Default.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node,
  displayTarget: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
}
Default.defaultProps = {
  display: 'block',
  element: 'div',
  displayTarget: 'belowTabletMax'
}

export {
  Default,
  Desktop,
  Laptop,
  Mobile,
  Tablet,
  TabletMax
}
