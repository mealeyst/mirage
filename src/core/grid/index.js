export {default as Grid} from './grid'
export {default as Sizer} from './sizer'
