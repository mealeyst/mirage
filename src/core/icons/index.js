//Icons
export {default as AIcon} from './AIcon'
export {default as BagIcon} from './BagIcon'
export {default as CardIcon} from './cards/CardIcon'
export {default as Checkmark} from './Checkmark'
export {default as Chevron} from './Chevron'
export {default as CircleChevron} from './CircleChevron'
export {default as GrayAIcon} from './GrayAIcon'
export {default as GrayLogo} from './GrayLogo'
export {default as Hamburger} from './Hamburger'
export {default as Logo} from './Logo'
export {default as NavyAIcon} from './NavyAIcon'
export {default as NavyLogo} from './NavyLogo'
export {default as Ruler} from './Ruler'
export {default as Spinner} from './Spinner'
export {default as WhiteAIcon} from './WhiteAIcon'
export {default as WhiteLogo} from './WhiteLogo'
export {default as XIcon} from './XIcon'
export {default as YellowCircleChevron} from './YellowCircleChevron'
export {default as GuaranteeIcon} from './GuaranteeIcon'
export {default as MailboxIcon} from './MailboxIcon'
