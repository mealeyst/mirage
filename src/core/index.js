export * from './grid'
export * from './icons'
export * from './image'
export * from './typography'
export * from './theme'
export * from './video'
