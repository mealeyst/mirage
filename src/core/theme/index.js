import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Fonts } from './fontFamilies'
import theme, { BaseFontSize } from './theme'

const ROATheme = (props) => {
  return (
    <main>
      <BaseFontSize />
      <Fonts />
      <ThemeProvider theme={theme} {...props} />
    </main>
  )
}

export default ROATheme
export { theme, ROATheme }
