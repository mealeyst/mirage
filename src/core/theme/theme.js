import { injectGlobal } from 'styled-components'
import { colors } from './colors'
import media, { breakpoints, breakpointsVerbose } from './mediaQueries'
import fontFamilies from './fontFamilies'
import gridSettings from './gridSettings'

const base = injectGlobal`
  html {
    font-size: 10px;
  }
`

export default {
  breakpoints: breakpoints,
  colors: colors,
  fonts: fontFamilies,
  grid: gridSettings,
  fixedPosition: 'relative',
  media: media,
  breakpointsVerbose: breakpointsVerbose,
  base: base
}
