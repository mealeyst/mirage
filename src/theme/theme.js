import { colors } from './colors'
import fontFamilies from './fontFamilies'

export default {
  colors: colors,
  fonts: fontFamilies,
  fixedPosition: 'relative'
}
