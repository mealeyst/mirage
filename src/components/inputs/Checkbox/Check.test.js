import React from 'react'
import { shallow } from 'enzyme'
import { css } from 'styled-components'
import 'jest-styled-components'

import { theme } from 'SRC/core/theme'
import StyledCheck, { checked } from './Check.base'

describe('(Base Component) Check', () => {

  test('matching the snapshot', () => {
    const component = shallow(<StyledCheck theme={theme} />)
    expect(component).toMatchSnapshot()
  })

  test('adding "checked" styles when checked is passes', () => {
    const component = shallow(<StyledCheck theme={theme} checked />)
    expect(component).toHaveStyleRule({
      modifier: css`${checked}`,
    })
  })
})
