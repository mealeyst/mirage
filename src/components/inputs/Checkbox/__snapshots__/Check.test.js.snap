// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Base Component) Check matching the snapshot 1`] = `
.c0 {
  fill: none;
  stroke-width: 20;
  stroke-linecap: round;
  stroke: #FFFFFF;
}

<Check
  className="c0"
/>
`;
